From 9367a52d289286dc76504f9e407e7a51fc5c0df1 Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@foss.st.com>
Date: Fri, 25 Feb 2022 10:16:25 +0100
Subject: [PATCH] ARM: dts: stm32: add vrefint support to adc on stm32mp13

Set STM32 ADC1&2 as consumers of BSEC, to retrieve
vrefint calibration data on STM32MP13x SoCs.

Change-Id: I1a0556445ef5875be42b53b7223e4cfa3d3d869d
Signed-off-by: Olivier Moysan <olivier.moysan@foss.st.com>
---
 arch/arm/boot/dts/st/stm32mp131.dtsi | 2 ++
 arch/arm/boot/dts/st/stm32mp133.dtsi | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/arch/arm/boot/dts/st/stm32mp131.dtsi b/arch/arm/boot/dts/st/stm32mp131.dtsi
index d2645f5a1869..97f589c15696 100644
--- a/arch/arm/boot/dts/st/stm32mp131.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp131.dtsi
@@ -837,6 +837,8 @@ adc2: adc@0 {
 				interrupts = <0>;
 				dmas = <&dmamux1 10 0x400 0x80000001>;
 				dma-names = "rx";
+				nvmem-cells = <&vrefint>;
+				nvmem-cell-names = "vrefint";
 				status = "disabled";
 
 				channel@13 {
diff --git a/arch/arm/boot/dts/st/stm32mp133.dtsi b/arch/arm/boot/dts/st/stm32mp133.dtsi
index df451c3c2a26..4a2a95a9a503 100644
--- a/arch/arm/boot/dts/st/stm32mp133.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp133.dtsi
@@ -56,6 +56,8 @@ adc1: adc@0 {
 				interrupts = <0>;
 				dmas = <&dmamux1 9 0x400 0x80000001>;
 				dma-names = "rx";
+				nvmem-cells = <&vrefint>;
+				nvmem-cell-names = "vrefint";
 				status = "disabled";
 
 				channel@18 {
-- 
2.39.5

